File: sample/src/main/java/me/xiaopan/assemblyadaptersample/bean/GameGroup.java
package me.xiaopan.assemblyadaptersample.bean;

import java.util.List;

import me.xiaopan.assemblyadapter.AssemblyGroup;

public class GameGroup implements AssemblyGroup {
    public String title;
    public List<Game> gameList;

    @Override
    public int getChildCount() {
        return gameList != null ? gameList.size() : 0;
    }

    @Override
    public Object getChild(int childPosition) {
        return gameList != null && childPosition < gameList.size() ? gameList.get(childPosition) : null;
    }
}

File: sample-old/src/main/java/com/github/panpf/assemblyadapter/sample/old/bean/GameGroup.kt
package com.github.panpf.assemblyadapter.sample.old.bean

import me.panpf.adapter.expandable.AssemblyGroup

class GameGroup : AssemblyGroup {
    var title: String? = null
    var gameList: ArrayList<Game>? = null

    override fun getChildCount(): Int {
        return if (gameList != null) gameList!!.size else 0
    }

    override fun getChild(childPosition: Int): Any? {
        return if (gameList != null && childPosition < gameList!!.size) gameList!![childPosition] else null
    }
}
